apis/bitbucketapi/apis.go

```diff
@@ -0,0 +1,83 @@
+package bitbucketapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/kubescape/go-git-url/apis"
+)
+
+const (
+	DEFAULT_HOST string = "bitbucket.org"
+)
+
+type IBitBucketAPI interface {
+	GetRepoTree(owner, repo, branch string, headers *Headers) (*Tree, error)
+	GetDefaultBranchName(owner, repo string, headers *Headers) (string, error)
+	GetLatestCommit(owner, repo, branch string, headers *Headers) (*Commit, error)
+}
+
+type BitBucketAPI struct {
+	httpClient *http.Client
+}
+
+func NewBitBucketAPI() *BitBucketAPI { return &BitBucketAPI{httpClient: &http.Client{}} }
+
+func (gl *BitBucketAPI) GetRepoTree(owner, repo, branch string, headers *Headers) (*Tree, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (gl *BitBucketAPI) GetDefaultBranchName(owner, repo string, headers *Headers) (string, error) {
+	id := owner + "%2F" + repo
+
+	body, err := apis.HttpGet(gl.httpClient, APIBranchingModel(id), headers.ToMap())
+	if err != nil {
+		return "", err
+	}
+
+	var data bitbucketBranchingModel
+	if err = json.Unmarshal([]byte(body), &data); err != nil {
+		return "", err
+	}
+
+	return data.Development.Name, nil
+}
+
+func (gl *BitBucketAPI) GetLatestCommit(owner, repo, branch string, headers *Headers) (*Commit, error) {
+	id := owner + "%2F" + repo
+
+	body, err := apis.HttpGet(gl.httpClient, APILastCommitsOfBranch(id, branch), headers.ToMap())
+	if err != nil {
+		return nil, err
+	}
+
+	var data Commits
+	err = json.Unmarshal([]byte(body), &data)
+	if err != nil {
+		return nil, err
+	}
+	return &data.Values[0], nil
+}
+
+// APIBranchingModel
+// API Ref: https://developer.atlassian.com/cloud/bitbucket/rest/api-group-branching-model/#api-group-branching-model
+// Example: https://bitbucket.org/!api/2.0/repositories/matthyx/ks-testing-public/branching-model
+func APIBranchingModel(id string) string {
+	return fmt.Sprintf("https://%s/!api/2.0/repositories/%s/branching-model", DEFAULT_HOST, id)
```

Comment:
It's nicer to use the `url` package, but it is not a must

Reply:
I see the need to escape variables from URL, so something like that instead?
```
	u := url.URL{
		Scheme: "https",
		Host:   DEFAULT_HOST,
		Path:   fmt.Sprintf("api/v4/projects/%s/repository/tree", id),
		RawQuery: fmt.Sprintf("ref=%s", branch),
	}
	return u.String()
```

Reply:
For the path, you should use the `url.Join` method